src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -52,17 +52,150 @@
 
 static const double NANOS_PER_SEC = 1000000000.0;
 
+#if !defined(__APPLE__)
+struct CPUTicks {
+  uint64_t usedTicks;
+  uint64_t totalTicks;
+};
+
+struct JVMTicks {
+  uint64_t userTicks;
+  uint64_t systemTicks;
+  CPUTicks cpuTicks;
+};
+
+struct PerfCounters {
+  int nProcs;
+  int stathz;		/* statistics clock frequency */
+  JVMTicks jvmTicks;
+  CPUTicks* cpus;
+};
+
+static int get_stathz(int *stathz) {
```

Comment:
A little bit more on the C centric code here in a C++ class.  It seems like if some of these static functions were instead private class methods on CPUPerformance then they could just access and manipulate member variables rather than pass them around so as parameters so much.  That, to me at least, would be better aligned with this as C++ code.